ingress/controllers/nginx/main.go

```diff
@@ -195,3 +188,28 @@ func handleSigterm(lbc *loadBalancerController) {
 	glog.Infof("Exiting with %v", exitCode)
 	os.Exit(exitCode)
 }
+
+const (
+	defTmpl      = "/etc/nginx/template/nginx.tmpl"
+	fallbackTmpl = "/etc/nginx/nginx.tmpl"
+)
+
+// checkTemplate verifies the NGINX template exists (file /etc/nginx/template/nginx.tmpl)
+// If the file does not exists it means:
+// a. custom docker image
+// b. standard image using watch-resource sidecar with emptyDir volume
```

Comment:
why emptyDir volume? isn't this just like the nginx example in main repo, except this sidecar is reusable with *any* existing pod that mounts a config map?